torchx/test/torchx/nx_test.exs

```diff
@@ -368,6 +368,57 @@ defmodule Torchx.NxTest do
       assert_equal(t, Nx.tensor([[0, 1, 2], [3, 4, 5]]))
     end
 
+    test "dot with vectors" do
+      t1 = Nx.tensor([1, 2, 3])
+      t2 = Nx.tensor([4, 5, 6])
+
+      out = Nx.dot(t1, t2)
+
+      assert_equal(out, Nx.tensor(32))
+    end
+
+    test "dot with matrices" do
+      t1 = Nx.tensor([[1, 2], [3, 4]])
+      t2 = Nx.tensor([[5, 6], [7, 8]])
+
+      out = Nx.dot(t1, t2)
+
+      assert_equal(out, Nx.tensor([[19, 22], [43, 50]]))
+    end
+
+    test "dot with vector and scalar" do
+      t = Nx.tensor([[1, 2, 3]])
+      assert Nx.shape(t) == {1, 3}
+      out = Nx.dot(t, 3)
+
+      assert Nx.shape(out) == {1, 3}
+      assert_equal(out, Nx.tensor([[3, 6, 9]]))
+    end
+
+    test "dot does not re-sort the contracting axes" do
```

Comment:
Could you add a test that shows that batching is not supported? 

This way when/if we support it, a new test will break and point towards adding more tests

Reply:
For sure! Just added it here https://github.com/elixir-nx/nx/pull/857/commits/3653d7a652767c633f61edc4af06ee374189fcf6